### DISCUSSION TITLE
Add role attribute to the extended valid elements config for all elements

### DISCUSSION THREAD

**Discussion by odonnella2** [umbraco/Umbraco-CMS#14118] [Features and ideas]
**What is it?**
Role is a global attribute with a focus on providing better web accessibility.

**Why?**
We have a number of clients focus on maintaining a certain WCAG compliance level. Role attributes will assist in [defining elements and relationship](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html) on elements entered into the WYSIWYG. 

**Tech Notes:**

Currently Umbraco uses tinymce's [extended_valid_elements](https://www.tiny.cloud/docs/configure/content-filtering/#extended_valid_elements) to enforce specific attributes which override any application specific configurations made. Recommendation is to add `role` to all elements.

```js
  //These are absolutely required in order for the macros to render inline
  //we put these as extended elements because they get merged on top of the normal allowed elements by tiny mce
  var extendedValidElements = "@[id|class|style],-div[id|dir|class|align|style],ins[datetime|cite],-ul[class|style],-li[class|style],-h1[id|dir|class|align|style],-h2[id|dir|class|align|style],-h3[id|dir|class|align|style],-h4[id|dir|class|align|style],-h5[id|dir|class|align|style],-h6[id|style|dir|class|align],span[id|class|style|lang],figure,figcaption";
```

This request is similar to https://github.com/umbraco/Umbraco-CMS/issues/10505

**Comment by iOvergaard** [comment]
This makes a lot of sense. I will mark this feature as up-for-grabs so please feel free to contribute with the proposed change!
